python/sglang/srt/layers/attention/utils.py

```diff
@@ -30,20 +101,20 @@ def create_flashinfer_kv_indices_triton(
         kv_start = tl.load(kv_start_idx + pid).to(tl.int32)
         kv_end = kv_start
     kv_end += tl.load(page_kernel_lens_ptr + pid).to(tl.int32)
-
-    num_loop = tl.cdiv(kv_end - kv_start, BLOCK_SIZE)
-    for i in range(num_loop):
-        # index into req_to_token_ptr needs to be int64
-        offset = tl.arange(0, BLOCK_SIZE).to(tl.int64) + i * BLOCK_SIZE
-        mask = offset < kv_end - kv_start
-        data = tl.load(
-            req_to_token_ptr
-            + req_pool_index * req_to_token_ptr_stride
-            + kv_start
-            + offset,
-            mask=mask,
-        )
-        tl.store(kv_indices_ptr + kv_indices_offset + offset, data, mask=mask)
+    
+    kv_range = kv_end - kv_start
+    num_pages = tl.cdiv(kv_range, PAGE_SIZE)
+    num_pages_loop = tl.cdiv(kv_range, BLOCK_SIZE)
+    base_ptr = req_to_token_ptr + req_pool_index * req_to_token_ptr_stride + kv_start 
+    for i in range(num_pages_loop):
+        page_offsets_int64 = (
+            tl.arange(0, NUM_PAGES_PER_BLOCK).to(tl.int64) + i * NUM_PAGES_PER_BLOCK
+            ) * PAGE_SIZE
+        out_page_offsets = tl.arange(0, NUM_PAGES_PER_BLOCK) + i * NUM_PAGES_PER_BLOCK
+        mask_page_range = page_offsets_int64 < (kv_range)
+        mask_out = out_page_offsets < num_pages 
+        data = tl.load(base_ptr + page_offsets_int64, mask=mask_page_range)
+        tl.store(kv_indices_ptr + kv_indices_offset + out_page_offsets, data // PAGE_SIZE, mask=mask_out)
```

Comment:
nit: I feel we could improve the naming a bit: 
```
page_offsets_int64 => token_offsets
out_page_offsets => page_offsets
mask_page_range => token_mask
mask_out => page_mask
num_pages_loop => num_loops
```

Reply:
Made the names better.